src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -219,9 +219,18 @@ public String getBranch() {
     }
 
     public void setBranch(String branch) {
+        updateDisplayName(branch);
         this.branch = branch;
     }
 
+    /**
+     * Update display name if default current working branch is being used.
+     * @param branch
+     */
+    public void updateDisplayName(String branch) {
+        this.displayName = displayName.substring(0, displayName.lastIndexOf('_') + 1) + branch;
```

Comment:
actually best solution would to be make the report generate the display name itself, so that we can even get rid of the display name as a whole, but we can leave it as it is for now.